### PR TITLE
Add decorator to class

### PR DESCRIPTION
**Pull request by amentetteh**
### PR raised for :
 - `Nameable` class created
 - `correct_name` that will raise a `NotImplementedError` implemented by  `Nameable`.
 - Person class turned to Nameable
 -  base Decorator prepared
 - CapitalizeDecorator and TrimmerDecorator prepared

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
```
Comment (MahmoudBakr23) [inline base_decorator.rb:6]: - [x] Please define the relationship between the Decorator and Nameable, it has to be done by using the accessor's attributes such as `attr_accessor :nameable`. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @amentetteh😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done on creating the correct decorating classes for this milestone so far
- Well done with the class (constructor) and custom methods
- Descriptive PR
- Correct Linters

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

- Please fix the MIT link for a professional README file

Cheers and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: base_decorator.rb (hunk 1 above)
Comment (amentetteh) [reply base_decorator.rb:6]: Ok. Thanks

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @amentetteh, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well in fixing the changes requested by the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**